internal/api/handler.go

```diff
@@ -341,8 +346,17 @@ func (h *Handler) handleTokenExchange(c *OAuthAppContext, w http.ResponseWriter,
 
 	h.metricsCollector.IncGeneratedToken(c.OAuthApplication.Identifier)
 
+	expiry := token.Extra("expires_in").(float64)
```

Comment:
Any reason to use `float64` here? Looking at the source code it is unmarshalled to `int32`.
From `oauth2` lib sources:
```go
type expirationTime int32

func (e *expirationTime) UnmarshalJSON(b []byte) error {
	if len(b) == 0 || string(b) == "null" {
		return nil
	}
	var n json.Number
	err := json.Unmarshal(b, &n)
	if err != nil {
		return err
	}
	i, err := n.Int64()
	if err != nil {
		return err
	}
	if i > math.MaxInt32 {
		i = math.MaxInt32
	}
	*e = expirationTime(i)
	return nil
}
```

Reply:
Typecasting to any type other than `float64` is causing a panic here. I get this error while trying to typecast it to `int32`:
```
http: panic serving 127.0.0.1:43024: interface conversion: interface {} is float64, not int32
```